MCprep_addon/materials/generate.py

```diff
@@ -954,7 +954,11 @@ def set_saturation_material(mat):
 			return  # requires regenerating material to add back
 		if len(desat_color) == 3:
 			desat_color += [1]  # add in alpha
-		sat_node.inputs[2].default_value = desat_color
+
+		if util.min_bv((3, 4)):
+			sat_node.inputs[6].default_value = desat_color
```

Comment:
On a spot check, shouldn't this be index 7? Index 6 is actually the "line in", ie the connected texture, while 7 would be the assignment of the actual color to apply.

```suggestion
			sat_node.inputs[7].default_value = desat_color
```

That being said, should we also consider using this new infra for version agnostic node selection? tbh i actually find your current code change easier to follow... but I guess if we want to be consistent we'd do something like below (95% sure this isn't quite right):

```
sat_node_in = get_node_socket(sat_node)
...
sat_node.inputs[sat_node_in[2]
```